src/codegen.cpp

```diff
@@ -1591,60 +1669,14 @@ emit_local_slot(int slot, jl_codectx_t *ctx)
 static Value*
 emit_temp_slot(int slot, jl_codectx_t *ctx)
 {
-    return emit_local_slot(slot + ctx->gc.argSpaceOffs, ctx);
+    Value *idx = ConstantInt::get(T_int32, slot);
+    return builder.CreateGEP(ctx->gc.tempSlot, idx);
 }
 
-// Create a GEP instruction for the @slot-th slot in the GC frame.
-// (without inserting it)
-static Instruction*
-create_local_slot(int slot, jl_codectx_t *ctx)
-{
-    jl_gcinfo_t *gc = &ctx->gc;
-#ifdef LLVM37
-    Instruction *ins =
-        GetElementPtrInst::Create(NULL, gc->argTemp,
-                                  ConstantInt::get(T_int32, slot));
-#else
-    Instruction *ins =
-        GetElementPtrInst::Create(gc->argTemp, ConstantInt::get(T_int32, slot));
-#endif
-    return ins;
-}
-
-// Create a GEP instruction for the @slot-th temporary variable in the
-// GC frame. (without inserting it)
-// The temporary variables are after all local variables in the GC frame.
-static Instruction*
-create_temp_slot(int slot, jl_codectx_t *ctx)
-{
-    return create_local_slot(slot + ctx->gc.argSpaceOffs, ctx);
-}
-
-static Value *make_gcroot(Value *v, jl_codectx_t *ctx, jl_sym_t *var)
+static Value *make_gcroot(Value *v, jl_codectx_t *ctx)
 {
     Value *froot = emit_temp_slot(ctx->gc.argDepth, ctx);
     builder.CreateStore(v, froot);
-#ifdef LLVM36
-    if (var != NULL) {
-        std::map<jl_sym_t *, jl_varinfo_t>::iterator it = ctx->vars.find(var);
-        if (it != ctx->vars.end() && ((llvm::MDNode*)it->second.dinfo) != NULL) {
-            if (ctx->debug_enabled) {
-                uint64_t slot = ctx->gc.argSpaceOffs + ctx->gc.argDepth;
-                SmallVector<int64_t, 9> addr;
-                addr.push_back(llvm::dwarf::DW_OP_plus);
-                addr.push_back(slot * sizeof(void*));
-                addr.push_back(llvm::dwarf::DW_OP_deref);
-#ifdef LLVM37
-                ctx->dbuilder->insertDeclare(ctx->gc.argTemp, it->second.dinfo,
-                    ctx->dbuilder->createExpression(addr),builder.getCurrentDebugLocation().get(),builder.GetInsertBlock());
-#else
-                ctx->dbuilder->insertDeclare(ctx->gc.argTemp, it->second.dinfo,
-                    ctx->dbuilder->createExpression(addr),builder.GetInsertBlock());
-#endif
-            }
-        }
-    }
-#endif
```

Comment:
I see.

Incidently, this is also the piece of debug info that is probably broken by #11508 and I'm just wondering what it is used for...